### PR TITLE
Gce round 3

### PR DESCRIPTION
**Pull request by justnoise**
Changes:
We've updated how configuration is handled.  Instead of supplying configuration for all clouds (a bit clunky if you just want to provision a single cloud), there is a separate configuration file for each cloud provider.

Updated to use `bookshelf/api_v3` cloud classes.

Where possible, global state has been moved into the bookshelf classes that represent a cloud instance.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,27 +27,43 @@ For EC2:
...
+    * OS_AUTH_SYSTEM (probably "rackspace")
 
-    * OS_AUTH_SYSTEM
 
-    * OS_REGION_NAME
+For Google Compute Engine:
 
-    * OS_NO_CACHE
+    GCE can support two authentication mechanisms.  For provisioning
+    locally, you'll want to use gcloud's authetication: `gcloud auth
```
Comment (wallrj) [inline README.md]: typo "authetication"

File: README.md [hunk 2]
```diff
@@ -28,27 +27,43 @@ For EC2:
...
+      use. If you have not uploaded a key pair to rackspace, your
+      public_key_filename will be uploaded to rackspace)
+
+    * RACKSPACE_PUBLIC_KEY_FILENAME (path to your public key)
 
-    * RACKSPACE_KEY_PAIR (the KEY_PAIR to use)
+    * RACKSPACE_PRIVATE_KEY_FILENAME (path to your public key)
 
-    * RACKSPACE_KEY_FILENAME (the full path to your .pem file)
+    * OS_USERNAME (your rackspace username. e.g. patton.oswalt)
```
Comment (Azulinho) [inline README.md]: I had to google who this guy was
'he is not funny'

File: ec2.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    us-west-2:
+      distribution:
+        centos7:
+          <<: *ec2_common
+          <<: *centos7_common
+          ami: 'ami-d440a6e7'
+        ubuntu1404:
+          <<: *ec2_common
+          <<: *ubuntu1404_common
+          ami: 'ami-22b9a343'
```
Comment (Azulinho) [inline ec2.yaml]: note for self.
need to figure out a way to scrape these out of the internet

File: fabfile.py [hunk 4]
```diff
@@ -45,6 +35,13 @@
 from tests.acceptance import acceptance_tests
 
 
+CLOUD_YAML_FILE = {
+    'gce': 'gce.yaml',
+    'ec2': 'ec2.yaml',
+    'rackspace': 'rackspace.yaml'
+}
+
```
Comment (Azulinho) [inline fabfile.py]: danka for this

File: fabfile.py [hunk 5]
```diff
@@ -121,114 +118,164 @@ def help():
           """)
```
Comment (Azulinho) [inline fabfile.py]: humm, reckon the help() might need to change as it was a just shamefull copy of the README contents (which have now changed)

File: lib/mycookbooks.py [hunk 6]
```diff
@@ -306,121 +276,33 @@ def load_config():
...
-        env.global_config = parse_config('config.yaml')['clouds']
-    except:
-        raise("Unable to parse config.yaml, see README")
-
-    # look up our state.json file, and override any settings found
-    load_state_from_disk()
+    # try:
+    #     env.global_config = parse_config('config.yaml')['clouds']
+    # except:
+    #     raise("Unable to parse config.yaml, see README")
```
Comment (Azulinho) [inline lib/mycookbooks.py]: delete

File: lib/mycookbooks.py [hunk 7]
```diff
@@ -306,121 +276,33 @@ def load_config():
...
 
+    # # look up our state.json file, and override any settings found
+    # load_state_from_disk()
```
Comment (Azulinho) [inline lib/mycookbooks.py]: and these

File: tests/acceptance.py [hunk 8]
```diff
@@ -136,7 +136,7 @@ def acceptance_tests_common_tests_for_flocker(distribution):
 
         # the client acceptance tests run on docker instances
         log_green('check that docker is running')
-        assert sudo('docker --version | grep "1.9."')
+        assert sudo('docker --version | grep "1.10."')
         assert process.is_up("docker")
```
Comment (Azulinho) [inline tests/acceptance.py]: ahhh, thanks for fixing that.
it got me today

**Comment by Azulinho** [comment]
awesome, thanks a million for this.
just minor comments.

now the difficult question,
how do you feel about squashing all those commits before merging?

**Comment by Azulinho** [comment]
maybe push it as a different branch so that we know its related to:
https://clusterhq.atlassian.net/browse/FLOC-4131

File: README.md (hunk 2 above)
Comment (justnoise) [inline README.md]: http://www.youtube.com/watch?v=r_DwZfyXAXI&t=0m55s

**Comment by justnoise** [comment]
https://github.com/ClusterHQ/CI-slave-images/pull/15 supersedes this PR.  Closing.

**Pull request closed without merging**